### ISSUE TITLE
High CPU usage with dispatch_in_background and MultiProcessingSubscriber

### ISSUE THREAD

**Issue by adam-urbanczyk** [getlogbook/logbook#246]
CPU usage close to 100%. I solved it by adding a `time.sleep(0.1)` call below this line: https://github.com/getlogbook/logbook/blob/11aa73ce49780235794f2a50dd1fe1c320e96183/logbook/queues.py#L276

Not sure if this is a correct solution though.

**Comment by vmalloc** [comment]
@adam-urbanczyk can you build a simple reproduction of this issue? There are a lot of details that probably matter to work out the reason for this.

**Comment by adam-urbanczyk** [comment]
@vmalloc  here is an example that allows to reproduce the issue. I confirm that my PR #247  solves it. The queue.get call needs to be blocking to actually respect the timeout argument.

```python
import multiprocessing as mp
import time
import sys
from logbook import Logger, StreamHandler, info
from logbook.queues import MultiProcessingHandler, MultiProcessingSubscriber

def dummy_process(q):
    
    proc_handler = MultiProcessingHandler(q)
    proc_handler.push_application()
    
    while True:
        info('Tick')
        time.sleep(.1)

if __name__ == '__main__':
    
    main_handler = StreamHandler(sys.stdout)
    main_handler.push_application()
    
    log_q = mp.Queue()
    
    proc = mp.Process(target=dummy_process,args=(log_q,))
    proc.start()
    
    sub = MultiProcessingSubscriber(log_q)
    sub.dispatch_in_background()
    
    while True:
        time.sleep(1)
```

**Comment by adam-urbanczyk** [comment]
Solved by #247
